### PR TITLE
feat(terraform): add CKV_NCP_1 about lb target group health check, CKV_NCP_2 about access control group description

### PR DESCRIPTION
**Pull request by Floodnut**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description
- Add new terraform provider Naver Cloud Platform.  
-- CKV_NCP_1 is associated with Loadbalancing target group's health check.
-- CKV_NCP_2 is associated with Access Control Group Description.

### Description
https://registry.terraform.io/providers/NaverCloudPlatform/ncloud/latest/docs/resources/lb_target_group
https://registry.terraform.io/providers/NaverCloudPlatform/ncloud/latest/docs/resources/access_control_group    

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/terraform/checks/resource/ncp/__init__.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import glob
+
+modules = glob.glob(join(dirname(__file__), "*.py"))
+__all__ = [basename(f)[:-3] for f in modules if isfile(f) and not f.endswith("__init__.py")]
+=======
+from pathlib import Path
+
+modules = Path(__file__).parent.glob("*.py")
+__all__ = [f.stem for f in modules if f.is_file() and not f.stem == "__init__"]
+>>>>>>> 5afbd2498e97b3c13edb7fe3f2c58291f4c4a2b2
```
Comment (gruebel) [inline checkov/terraform/checks/resource/ncp/__init__.py:12]: looks like some merge conflict 😄 please use the lower one 👍 
```suggestion
from pathlib import Path

modules = Path(__file__).parent.glob("*.py")
__all__ = [f.stem for f in modules if f.is_file() and not f.stem == "__init__"]
```

File: checkov/terraform/checks/resource/ncp/AccessControlGroupRuleDescription.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        if rule_type == 'group_or_rule_description':
+            if 'description' in conf.keys():
+                self.evaluated_keys = ['description']
+                if conf['description']:
+                    return CheckResult.PASSED
+            return CheckResult.FAILED
+
+        if rule_type in conf.keys():
+            for rule in conf[rule_type]:
+                if isinstance(rule, dict) and ('description' not in rule.keys() or not rule['description']):
```
Comment (gruebel) [inline checkov/terraform/checks/resource/ncp/AccessControlGroupRuleDescription.py:41]:
```suggestion
                if isinstance(rule, dict) and rule.get('description'):
```
does the same with less code 🙂

File: checkov/terraform/checks/resource/ncp/AccessControlGroupRuleDescription.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            return CheckResult.FAILED
+
+        return group_result
+
+    def check_rule(self, rule_type, conf):
+        if rule_type == 'group_or_rule_description':
+            if 'description' in conf.keys():
+                self.evaluated_keys = ['description']
+                if conf['description']:
+                    return CheckResult.PASSED
```
Comment (gruebel) [inline checkov/terraform/checks/resource/ncp/AccessControlGroupRuleDescription.py:36]:
```suggestion
            self.evaluated_keys = ['description']
            if conf.get('description'):
                return CheckResult.PASSED
```

**Review by gruebel — COMMENTED** [review]
nice job 🥇 , can you adjust the tests to have dedicated TF files and a test file to simulate an actual run. You can check this PR on how it should look like. If you have problems, let me know, then I can help you out 🙂

**Comment by pj991207** [comment]
> n you adjust the tests to have dedicated TF files and a test file to simulate an actual run.

did you say that making test code like this snapshot?
image
image

**Comment by pj991207** [comment]
> that making test code like this snapshot?

I made a test code like that 
image
image

**Comment by gruebel** [comment]
> > n you adjust the tests to have dedicated TF files and a test file to simulate an actual run.
> 
> did you say that making test code like this snapshot? image image

yeah correct. how you wrote the tests is the old way of doing it and it resulted in some issues we had. New tests should use the style I described and over time we update the old tests to fit the newer style.

**Comment by Floodnut** [comment]
We change our test codes with CONTRIBUTING.md styles.
Can you check we commited and what we missed?
thx.

**Review by gruebel — COMMENTED** [review]
thanks looks great. would be nice, if you can the mentioned resource to be tested too 🙂

File: tests/terraform/checks/resource/ncp/example_AccessControlGroupRuleDescription/main.tf [hunk 4]
```diff
@@ -0,0 +1,10 @@
+resource "ncloud_access_control_group" "pass" {
```
[the remark this answers is not in the document]
Comment (Floodnut) [reply tests/terraform/checks/resource/ncp/example_AccessControlGroupRuleDescription/main.tf:1]: I'll check additional resource `ncloud_access_control_group_rule`!! thanks!

**Review by gruebel — APPROVED** [review]
great work 🍻

**Review by Saarett — APPROVED** [review]

**Pull request merged**